redisson/src/main/java/org/redisson/client/handler/CommandDecoder.java

```diff
@@ -98,12 +103,6 @@ protected final void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object>
                 }
             }
         } else {
-            if (holder.getChannelPromise().isDone() && !holder.getChannelPromise().isSuccess()) {
-                sendNext(ctx.channel());
-                // throw REPLAY error
-                in.indexOf(Integer.MAX_VALUE/2, Integer.MAX_VALUE, (byte) 0);
-                return;
-            }
```

Comment:
@mrniko could you review this, I think if the holder's promise is fail, it should decode with next command immediately.

Reply:
this is to avoid recursion in case of many failed command

Reply:
@mrniko The method ```sendNext``` skip fail command, then try to decode with next command in the command queue, in my test case it can also fixed ```RedisResponseTimeoutException```.
```
            if (holder.getChannelPromise().isDone() && !holder.getChannelPromise().isSuccess()) {
                sendNext(ctx.channel());
                decode(ctx, in, out);
                return;
            }
```

Reply:
@zhwq1216 

Let me clarify. On REPLAY error netty tries to decode command once more, but without recursion.

Reply:
Ok, I understand. Thank you!